modules/chapter3/pages/section1.adoc

```diff
@@ -1,3 +1,142 @@
+
+The exercise bellow covers cells in more detail.
+
+== Kernels
+
+In the context of JuypterLab (and Jupyter Notebook), a kernel is the process that runs the code cells in a notebook.
+A kernel determines the language that you can execute in a notebook, which by default is Python.
+
+[NOTE]
+====
+Kernels do not execute markdown cells.
+====
+
+JupyterLab includes a Python kernel, called https://docs.jupyter.org/en/latest/projects/kernels.html#term-ipykernel[ipykernel].
+This is the default kernel in JuypterLab, and depends on IPython, an interactive Python shell.
+
+If you wish to create notebooks that contain code cells in other languages, you can explore the https://github.com/jupyter/jupyter/wiki/Jupyter-kernels[Kernels List].
+Many of these kernels are actively maintained by the Juypter community.
+
+You can interact with kernels without JupyterLab to create your own notebook environments.
+For example, the Python extension of VSCodium supports Juypter notebooks without installing JuypterLab.
+Instead, VSCodium interacts with ipykernel.
+
+
+== Exercise: Exploring Juypter Notebooks
+
+To continue learning about notebooks, open RHODS.
+
+1. In a new or an existing data science project, create a `Minimal Python` workbench.
+
+2. Open the notebook and clone the https://github.com/RedHatQuickCourses/rhods-qc-apps repository.
+
+3. Open the `1.intro/chapter3/notebooks-intro.ipynb` notebook and follow the instructions.
+
+
+== Exercise: Opening Terminals
+
+You can also open terminals into the workbench container via JupyterLab.
+
+1. Open the workbench from the previous exercise.
+You can also create a new one.
+
+2. On the btn:[Launcher] tab, click the btn:[Terminal] card.
+If the launcher tab is not open, you can open a new launcher by clicking the btn:[+] tab.
++
+image::jupyter-terminal.png[]
+
+3. A new terminal opens:
++
+image::jupyter-terminal-open.png[]
+
+
+== Working with Git
+
+You have previously used JuypterLab to clone repositories in your workbench.
+You can also commit and push your changes to your remote repository.
+
+After you have cloned the repository, the Git tab in JuypterLab displays information about the current repository, including the branch, the current changes and the history.
+
+image::jupyter-git.png[]
+
+If you make any changes, the Git card displays them:
+
+image::jupyter-git-changes.png[]
+
+You can use the btn:[+] icon in this area to stage your changes.
+
+image::jupyter-git-stage.png[]
+
+You can enter the commit message and the description at the bottom, and click btn:[Commit].
+At this point, JupyterLab might prompt you to enter your name and email.
+
+To push (and pull) changes, use the cloud icons at the top.
+
+image::jupyter-git-push.png[]
+
+At this point, JupyterLab might prompt you to enter your Git credentials.
+
+image::jupyter-git-credentials.png[]
+
+[NOTE]
+====
+You can also work with Git by opening a terminal in JuypterLab and using the `git` CLI.
+====
```

Comment:
Ted left a comment in this section of the design - 

> Somewhere... maybe here? Mention adding/deleting cells, running cells selectively. Just to name a few: You can display images, e.g. data visualizations, or even play sound clips.

Reply:
What about Errors and debugging - one or two lines about how to debug?

Reply:
These are addressed in the accompanying `rhods-qc-apps` notebook: 

https://github.com/RedHatQuickCourses/rhods-qc-apps/blob/f95245a722818156858ce58650228b386270d8ae/1.intro/chapter3/intro/notebooks-intro.ipynb